### DISCUSSION TITLE
Excited About Commonware – A Few Foundational Questions

### DISCUSSION THREAD

**Discussion by OChOK-crypto** [commonwarexyz/monorepo#272] [Q&A] [answered]
I don’t have any coding skills, and my English isn’t very strong, but I’m trying to get a clearer, more intuitive understanding of Commonware and its goals. I have two quick questions:

1.Is Commonware a Layer 0 platform that enables developers to build anything from Layer 1s to applications, offering flexibility for various use cases?

2.Am I right in thinking that Commonware focuses on collaboration rather than disrupting existing infrastructure, even with ecosystems like Ethereum, IBC, or Substrate?

If you could share any specific examples or insights to help me understand this better, I’d greatly appreciate it!

**Comment by patrick-ogrady** [comment, marked as answer]
Great questions!

> 1. Is Commonware a Layer 0 platform that enables developers to build anything from Layer 1s to applications, offering flexibility for various use cases?

Commonware is not a L0, L1, L2, etc. At this point, it is a library of "state-of-the-art" primitives for building blockchains/AVSes/etc. We call these primitives "commonware" because they are open source (MIT/Apache-2) and not affiliated with a particular ecosystem.

> 2. Am I right in thinking that Commonware focuses on collaboration rather than disrupting existing infrastructure, even with ecosystems like Ethereum, IBC, or Substrate?

The primitives found in the Commonware Library can be used on their own to create something from scratch or added to existing blockchains. It is up to the developer that wants to use the software we're working on here as they see fit.

> If you could share any specific examples or insights to help me understand this better, I’d greatly appreciate it!

Check out [examples](https://github.com/commonwarexyz/monorepo/tree/main?tab=readme-ov-file#examples) for a collection of different "toys" we created to show the different ways you could employ the Commonware Library (only one of them uses consensus like a traditional blockchain).

Hope that helps!

**Comment by OChOK-crypto** [comment]
Thank you for your reply!

I’ll admit, due to my lack of knowledge, I couldn’t fully grasp the content of the links you shared. But I can clearly feel your deep passion and thoughtful analysis in tackling challenges in the blockchain space.

This has inspired me to learn more and deepen my understanding.

Wishing you all the best with the project, and thanks again!
